app/js/components/tabs/tabcomponents/encounterComponent.jsx

```diff
@@ -517,7 +519,7 @@ class EncounterComponent extends Component {
 
             <div className="form-group">
               <div className="col-sm-offset-2 col-sm-6">
-                <button type="submit" className="btn btn-success">Search</button>
+                <button type="submit" className="btn btn-success" onClick={this.searchByLocation}>Search</button>
```

Comment:
What's drawback does calling `this.searchByLocation` from form submit event vs submit button click event have?

Reply:
When you call the method from form submit, you allow for implicit form submission. i.e., when a user focusses on the form and presses the `Enter key`, the form will be submitted. 

There is however not much difference between the two methods, except that, in my current implementation, should in future another submit button be added to  the form, it would not be fired for any other function aside from what is delegated to it. Also, should the button `type` change, it would still serve the purpose for which it was created. 

When handling the submit on the form, usually, when the Enter key is pressed, it seeks out the first ‘submit button’ inside the form and uses it to  submit the form.  This may result in unusual  behaviour. 

Thus, as a personal preference,  I opt to explicitly define the purpose of my buttons, so  that they trigger the correct functions when clicked on. Which I figure should someone desire to maybe add the functionality of creating an option where the current form, leads to  a new form within it whose values are dependent on the current form `i.e..` use of `next` button. They can manipulate that within the form, but at the end of the day,  the Search button will  always take everything within that form method, and handle the submission of that parent form. 

I  hope that explains the logic behind this implementation.  Feel  free to  ask for clarification and also share your own thoughts in this regard.

Reply:
👍